plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-crud-controller.php

```diff
@@ -176,6 +176,43 @@ protected function save_object( $request, $creating = false ) {
 		}
 	}
 
+	protected function duplicate_product_cleanup( $sku ) {
+		global $wpdb;
+	
+		$time_threshold = current_time( 'mysql' , 1) - 5;
+	
+		// Fetch IDs of products with the specified SKU created within the last 5 seconds
+		$query = $wpdb->prepare("
+			SELECT p.ID
+			FROM $wpdb->posts p
+			INNER JOIN $wpdb->postmeta pm ON p.ID = pm.post_id
+			WHERE pm.meta_key = '_sku' AND pm.meta_value = %s
+			AND p.post_type = 'product' AND p.post_date_gmt >= %s
+			ORDER BY p.post_date_gmt DESC
+		", $sku, date( 'Y-m-d H:i:s', $time_threshold ));
+	
+		$product_ids = $wpdb->get_col( $query );
+	
+		if (empty( $product_ids )) {
+			return false;
+		}
+	
+		array_shift( $product_ids );
+	
+		if (!empty( $product_ids )) {
+			$ids_to_delete = implode(',', array_map( 'intval', $product_ids ));
+	
+			// Delete prodocts post
+			$wpdb->query( "DELETE FROM $wpdb->posts WHERE ID IN ($ids_to_delete)" );
+	
+			// Delete products post meta
+			$wpdb->query( "DELETE FROM $wpdb->postmeta WHERE post_id IN ($ids_to_delete)" );
```

Comment:
Is it safe here to do the product deletions with raw SQL? This will bypass actions/filters that run during normal programmatic production deletion, such as [`woocommerce_pre_delete_product`](https://github.com/woocommerce/woocommerce/blob/db50513c42d77e533f76c4b9b31e42b8709f824b/plugins/woocommerce/includes/abstracts/abstract-wc-data.php#L207-L207) and [`woocommerce_delete_product`](https://github.com/woocommerce/woocommerce/blob/db50513c42d77e533f76c4b9b31e42b8709f824b/plugins/woocommerce/includes/data-stores/class-wc-product-data-store-cpt.php#L303).